### PR TITLE
Test for Windows encoding issue

### PR DESCRIPTION
**Pull request by ken2812221**
This PR removes unused Windows compatible path. Encoding issues has been solved.

### CODE REVIEW

**Comment by jnewbery** [comment]
suggest you update the title to describe what's being tested (rather than referencing a different issue)

**Comment by sipsorcery** [comment]
Without this PR bitcoind does not get very far with a unicode directory (the conf file uses `datadir=C:\Temp\资金钱包\Bitcoin`) :

> >bitcoind -conf=c:\temp\bitcoin.conf
Error reading configuration file: specified data directory "C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin" does not exist.

After applying this PR bitcoind gets to the point where it attempts to open the wallet file:

> >bitcoind -conf=c:\temp\bitcoin.conf
2018-08-31T11:22:29Z Bitcoin Core version v1.17.99.0-unk (release build)
2018-08-31T11:22:30Z InitParameterInteraction: parameter interaction: -whitelistforcerelay=1 -> setting -whitelistrelay=1
2018-08-31T11:22:30Z Validating signatures for all blocks.
2018-08-31T11:22:30Z Setting nMinimumChainWork=0000000000000000000000000000000000000000000000000000000000000000
2018-08-31T11:22:30Z Using the 'standard' SHA256 implementation
2018-08-31T11:22:31Z Default data directory C:\Users\aaron\AppData\Roaming\Bitcoin
2018-08-31T11:22:31Z Using data directory C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest
2018-08-31T11:22:31Z Using config file c:\temp\bitcoin.conf
....
2018-08-31T11:22:32Z Using wallet wallet.dat
2018-08-31T11:22:32Z BerkeleyEnvironment::Open: LogDir=C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\database ErrorFile=C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\db.log
2018-08-31T11:22:32Z BerkeleyEnvironment::Open: Error -30974 opening database environment: DB_RUNRECOVERY: Fatal error, run database recovery
2018-08-31T11:22:32Z Moved old C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\database to C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\database.1535714552.bak. Retrying.
2018-08-31T11:22:32Z BerkeleyEnvironment::Open: LogDir=C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\database ErrorFile=C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets\db.log
2018-08-31T11:22:32Z BerkeleyEnvironment::Open: Error -30974 opening database environment: DB_RUNRECOVERY: Fatal error, run database recovery
2018-08-31T11:22:32Z Error: Error initializing wallet database environment "C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets"!
Error: Error initializing wallet database environment "C:\Temp\ÞÁäÚçæÚÆ▒Õîà\Bitcoin\regtest\wallets"!

I'll work out how to apply the leveldb patch and retry.

**Comment by ken2812221** [comment]
@sipsorcery Thanks for testing. Could you test this again? This shall not contain garbled text.

**Comment by sipsorcery** [comment]
@ken2812221 I'm testing with a Visual Studio build (currently having difficulty running the mingw32 build to run due to something being up with libzmq; still tracking that one down).

I've added leveldb to my VS solution and patched it. Bitcoind now seems to work correctly with unicode characters in the data directory path IF the wallet is disabled. If the wallet is enabled then the BerkleyDB library is failing to initialise.

> 2018-09-01T07:37:21Z BerkeleyEnvironment::Open: LogDir=C:\Temp\资金钱包\Bitcoin\regtest\wallets\database ErrorFile=C:\Temp\资金钱包\Bitcoin\regtest\wallets\db.log
2018-09-01T07:37:21Z BerkeleyEnvironment::Open: Error -30974 opening database environment: DB_RUNRECOVERY: Fatal error, run database recovery
2018-09-01T07:37:21Z Moved old C:\Temp\资金钱包\Bitcoin\regtest\wallets\database to C:\Temp\资金钱包\Bitcoin\regtest\wallets\database.1535787441.bak. Retrying.
2018-09-01T07:37:21Z BerkeleyEnvironment::Open: LogDir=C:\Temp\资金钱包\Bitcoin\regtest\wallets\database ErrorFile=C:\Temp\资金钱包\Bitcoin\regtest\wallets\db.log
2018-09-01T07:37:21Z BerkeleyEnvironment::Open: Error -30974 opening database environment: DB_RUNRECOVERY: Fatal error, run database recovery
2018-09-01T07:37:23Z Error: Error initializing wallet database environment "C:\Temp\资金钱包\Bitcoin\regtest\wallets"!
Error: Error initializing wallet database environment "C:\Temp\2018-09-01T07:37:44Z Interrupting HTTP RPC server
2018-09-01T07:37:44Z Interrupting RPC

Kudos for working on this! Unicode and Windows is a minefield. Have you seen http://utf8everywhere.org/ (Chapter 10 is dedicated to UTF on Windows).

**Comment by ken2812221** [comment]
You can see #13884 , I pre-defined UNICODE and _UNICODE. You should define them while you are compiling berkeleydb. But I don't know how to do that with vcpkg.

**Comment by sipsorcery** [comment]
I've built this PR on Ubuntu and Windows Subsystem for Linux (Ubuntu 18) and bitcoind executes correctly with a data directory that contains unicode characters. 

I am not familiar with nsis and have not tested that part of the PR. How do I use nsis to the build the installer package?

One additional task that would need to be done after this PR is merged is to update the [Windows build instructions](https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md) to add the new packages:

` sudo apt install scons wine-binfmt wine64 g++-mingw-w64-i686`

File: test/functional/test_runner.py [hunk 1]
```diff
@@ -244,10 +244,6 @@ def main():
     # Create base test directory
     tmpdir = "%s/test_runner_₿_🏃_%s" % (args.tmpdirprefix, datetime.datetime.now().strftime("%Y%m%d_%H%M%S"))
 
-    # If we fixed the command-line and filename encoding issue on Windows, these two lines could be removed
-    if config["environment"]["EXEEXT"] == ".exe":
-        tmpdir = "%s/test_runner_%s" % (args.tmpdirprefix, datetime.datetime.now().strftime("%Y%m%d_%H%M%S"))
```
Comment (MarcoFalke) [inline test/functional/test_runner.py]: Nice

File: .appveyor.yml [hunk 2]
```diff
@@ -7,12 +7,11 @@ environment:
   APPVEYOR_SAVE_CACHE_ON_ERROR: true
   CLCACHE_SERVER: 1
   PACKAGES: boost-filesystem boost-signals2 boost-test libevent openssl zeromq berkeleydb secp256k1 leveldb
-  PYTHONIOENCODING: utf-8
+  PATH: 'C:\Python37-x64;C:\Python37-x64\Scripts;%PATH%'
+  PYTHONUTF8: 1
```
Comment (ken2812221) [inline .appveyor.yml]: Upgrade to python 3.7 because this require PEP540, make it possible to use utf-8 filename

File: .appveyor.yml [hunk 3]
```diff
@@ -21,7 +20,8 @@ install:
 - ps:  for ($i=0; $i -lt $packages.length; $i++) {
            $env:ALL_PACKAGES += $packages[$i] + ":" + $env:PLATFORM + "-windows-static "
        }
-- cmd: git -C C:\Tools\vcpkg pull # This is a temporary fix, can be removed after appveyor update its image to include Microsoft/vcpkg#4046
+- cmd: git -C C:\Tools\vcpkg fetch https://github.com/ken2812221/vcpkg.git unicode-branch
+- cmd: git -C C:\Tools\vcpkg checkout FETCH_HEAD
```
Comment (ken2812221) [inline .appveyor.yml]: This is just for testing that apply utf-8 filename changes, it would have to change upstream.

**Comment by laanwj** [comment]
Looks like most of the prerequisites have been merged now, can we make this in before the 0.18 split?

**Comment by ken2812221** [comment]
Appveyor is going to update the images in a few days. After then, we don't have to get the latest vapkg port data to get Berkeley db package and re-build vcpkg every time (3+ minutes). I will rebase it once they update the CI image.

**Comment by laanwj** [comment]
Ok, just be aware that the split is going to be 2019-03-01 and it's going to be busy around then.

**Comment by ken2812221** [comment]
This is ready to review/merge. I have already rebased it.

**Pull request merged**